Bug fixes + minor doc site improvements

slhinyc opened the pull request:
## What's in this PR
- Update `text` variant of `sl-button` to strip all padding so that it aligns better with other elements on the screen ([Link in review app](https://shoelace-kdloz39o7-teamshares.vercel.app/components/button#text-buttons))
- Add Link page for documentation of default link style and link button options ([Link in review app](https://shoelace-kdloz39o7-teamshares.vercel.app/teamshares/link))
- Fix `sl-checkbox` and `sl-radio` bug where a `contained` checkbox or radio had a hover state even when disabled ([Link in review app](https://shoelace-kdloz39o7-teamshares.vercel.app/components/checkbox#contained))
- Fix padding cascade bug for `sl-dialog` body for `announcement` variant ([Link in review app](https://shoelace-kdloz39o7-teamshares.vercel.app/components/dialog#announcement-dialog))
- Fix `sl-button` bug where a disabled link (`a`) button still opens the `href` link ([Link in review app](https://shoelace-jc5m8xsbk-teamshares.vercel.app/components/button#link-buttons))
- Misc doc site improvements
  - Update props tables to be more readable with no scrolling
  - Add "no footer" and default width examples to `sl-dialog` docs
  - Fix bug where Event names are hidden when `SLIM` preview is selected
  - Fix bug where icon doesn't display for component properties where `reflect` is `true`

---

### Screenshots of props table update

**Old props table**
Full table isn't visible without horizontal scroll
<img width="866" alt="Screenshot 2024-07-17 at 14 32 44" src="https://github.com/user-attachments/assets/dd9c382e-36da-49ac-9b10-770f3c8be52b">

**Updated props table**
Updated styling & layout so that no scrolling is needed to see the full table
<img width="859" alt="Screenshot 2024-07-17 at 14 44 24" src="https://github.com/user-attachments/assets/a55c502f-1e55-4a2d-b160-d49af66184eb">

kdonovan commented on docs/assets/styles/docs.css:
```diff
@@ -682,20 +684,36 @@ table td p:last-child {
+  width: 100%;
+}
+
+/* .table-scroll code {
+  white-space: nowrap;
+} */
+
```
Should we just cut this?

slhinyc replied:
Ah, good call -- just deleted!

kdonovan commented on docs/pages/components/dialog.md:
```diff
@@ -269,6 +280,15 @@ const App = () => {
+| Dialog variant | Button variant | Icon name |
+| ----------------------- | -------------- | ---------------------- |
+| `warning` | `warning` | `exclamation-triangle` |
+| `default` confirmation | `primary` | `exclamation-circle` |
+| `default` informational | `primary` | `info-circle` |
```
Should the `default`s still be in there?

slhinyc replied:
Yes, they are two separate types of dialogs that should use the `default` variant. But your comment made me realize the table could be laid out more clearly, so I made an update! 

<img width="843" alt="Screenshot 2024-07-17 at 15 22 28" src="https://github.com/user-attachments/assets/8d2cfb6c-9f26-4f11-87ca-dac45f80344f">

kdonovan reviewed (approved):
🙌

slhinyc commented:
@kathleenteamshares @kdonovan -- Could you both take a look at some recent additions to this PR?

There was an additional component update that we needed to make & rather than cut this patch release first and open another PR, I added to this PR so that we can just wrap all the changes in a new minor release.

Updated the main PR notes to reflect all the changes. TY both! 🙏

kdonovan reviewed (approved):
🚗
